lib/src/over_react_redux/over_react_redux.dart

```diff
@@ -0,0 +1,393 @@
+      if (result is UiProps) result = (result as UiProps).props;
+      return jsBackingMapOrJsCopy(result);
+    }
+
+    JsMap handleMergeProps(JsMap jsStateProps, JsMap jsDispatchProps, JsMap jsOwnProps) {
+      final stateProps = factory(new JsBackedMap.backedBy(jsStateProps));
+      final dispatchProps = factory(new JsBackedMap.backedBy(jsDispatchProps));
+      final ownProps = factory(new JsBackedMap.backedBy(jsOwnProps));
+      var result = mergeProps(stateProps, dispatchProps, ownProps);
+
+      // Pull out the underlying backing map to avoid copying where possible
+      if (result is UiProps) result = (result as UiProps).props;
+      return jsBackingMapOrJsCopy(result);
+    }
+
+    bool handleAreStatesEqual(ReactInteropValue jsNext, ReactInteropValue jsPrev) =>
+        areStatesEqual(unwrapInteropValue(jsNext), unwrapInteropValue(jsPrev));
+
+    bool handleAreOwnPropsEqual(JsMap jsNext, JsMap jsPrev) =>
+        areOwnPropsEqual(factory(new JsBackedMap.backedBy(jsNext)), factory(new JsBackedMap.backedBy(jsPrev)));
+
+    bool handleAreStatePropsEqual(JsMap jsNext, JsMap jsPrev) =>
+        areStatePropsEqual(factory(new JsBackedMap.backedBy(jsNext)), factory(new JsBackedMap.backedBy(jsPrev)));
+
+    bool handleAreMergedPropsEqual(JsMap jsNext, JsMap jsPrev) =>
+        areMergedPropsEqual(factory(new JsBackedMap.backedBy(jsNext)), factory(new JsBackedMap.backedBy(jsPrev)));
+
+    final hoc = mockableJsConnect(
+      mapStateToProps != null ? allowInterop(handleMapStateToProps) : mapStateToPropsWithOwnProps != null ? allowInterop(handleMapStateToPropsWithOwnProps) : null,
+      mapDispatchToProps != null ? allowInterop(handleMapDispatchToProps) : mapDispatchToPropsWithOwnProps != null ? allowInterop(handleMapDispatchToPropsWithOwnProps) : null,
+      mergeProps != null ? allowInterop(handleMergeProps) : null,
+      new JsConnectOptions(
+        areStatesEqual: allowInterop(handleAreStatesEqual),
+        areOwnPropsEqual: allowInterop(handleAreOwnPropsEqual),
+        areStatePropsEqual: allowInterop(handleAreStatePropsEqual),
+        areMergedPropsEqual: allowInterop(handleAreMergedPropsEqual),
+        forwardRef: forwardRef,
+        pure: pure,
+        context: context?.jsThis ?? JsReactRedux.ReactReduxContext,
+      ),
+    )(dartComponentClass);
+
+    final hocJsFactoryProxy = new ReactJsComponentFactoryProxy(hoc, shouldConvertDomProps: false, alwaysReturnChildrenAsList: true);
+
+    TProps connectedFactory([Map props]) {
+      return (factory(props)..componentFactory = hocJsFactoryProxy);
+    }
+
+    return connectedFactory;
+  }
+
+  return wrapWithConnect;
+}
+
+bool _defaultEquality(Object a, Object b) => a == b;
+bool _shallowMapEquality(Object a, Object b) => const MapEquality().equals(a, b);
+
+@JS('ReactRedux.connect')
+external ReactClass Function(ReactClass) _jsConnect(
+    [
+      Function mapStateToProps,
+      dynamic mapDispatchToProps,
+      dynamic mergeProps,
```

Comment:
but i typed `mergeProps` as `Function` as well. an additional reason for the loose typing is that `mapStateToProps` and `mapDispatchToProps` both have alternate methods that could be used in dart `mapStateToPropsWithOwnProps` that has an additional 2nd argument. In the react-redux docs this will change its behavior and could cause additional calls/renders. https://react-redux.js.org/using-react-redux/connect-mapstate#return-values-determine-if-your-component-re-renders